File: Leetcode Weekly Contests/Weekly-Contest-380/Count Elements with Maximum Frequency/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int maxFrequencyElements(vector<int>& nums) {
        int n = nums.size();
        map<int,int>mp;
        for(int i=0; i<n; i++){
            mp[nums[i]]++;
        }
        int mxFreq = 0;
        for(auto x:mp){
            mxFreq = max(mxFreq, x.second);
        }
        int ans = 0;
        for(auto x: mp){
            if(x.second == mxFreq){
                ans += x.second;
            }
        }
        return ans;
    }
};

File: Leetcode Weekly Contests/Weekly-Contest-380/Count Elements with Maximum Frequency/Solution.java
class Solution {
    public int maxFrequencyElements(int[] nums) {
        int[] freq = new int[101];
        int maxi = 0;
        
        for(int num : nums) {
            freq[num]++;
            maxi = Math.max(maxi, freq[num]);
        }
        
        int ans = 0;
        
        for(int el : freq) {
            if(el == maxi) ans += el;
        }
        
        return ans;
    }
}